Android/app/src/main/java/app/intra/DnsVpnAdapter.java

```diff
@@ -60,30 +69,140 @@
 
   private static final int DNS_DEFAULT_PORT = 53;
 
+  // VPN parameters
+  private static final int VPN_INTERFACE_MTU = 32767;
+  // Randomly generated unique local IPv6 unicast subnet prefix, as defined by RFC 4193.
+  private static final String IPV6_SUBNET = "fd66:f83a:c650::%s";
+
   // Misc constants
   private static final int SLEEP_MILLIS = 1500;
 
   private final @NonNull
   DnsVpnService vpnService;
-  private final ParcelFileDescriptor tunFd;
+  @NonNull private final ParcelFileDescriptor tunFd;
   private final FileInputStream in;
   private final FileOutputStream out;
 
-  // Owners can safely mutate serverConnection at any time, including setting it to null.
-  @Nullable
-  ServerConnection serverConnection = null;
+  static DnsVpnAdapter get(DnsVpnService vpnService) {
```

Comment:
Could you rename this method to indicate that it establishes the VPN?

Reply:
Good idea.  Renamed from `get` to `establish`.